src/views-components/project-copy-dialog/project-copy-dialog.tsx

```diff
@@ -1,29 +1,53 @@
 // Copyright (C) The Arvados Authors. All rights reserved.
 //
 // SPDX-License-Identifier: AGPL-3.0
-import { Dispatch, compose } from "redux";
-import { withDialog } from "~/store/dialog/with-dialog";
-import { dialogActions } from "~/store/dialog/dialog-actions";
-import { ProjectCopy, CopyFormData } from "~/components/project-copy/project-copy";
-import { reduxForm, startSubmit, stopSubmit, initialize } from 'redux-form';
-import { resetPickerProjectTree } from "~/store/project-tree-picker/project-tree-picker-actions";
 
-export const PROJECT_COPY_DIALOG = 'projectCopy';
-export const openProjectCopyDialog = (data: { projectUuid: string, name: string }) =>
-    (dispatch: Dispatch) => {
-        dispatch<any>(resetPickerProjectTree());
-        const initialData: CopyFormData = { name: `Copy of: ${data.name}`, projectUuid: '', uuid: data.projectUuid };
-        dispatch<any>(initialize(PROJECT_COPY_DIALOG, initialData));
-        dispatch(dialogActions.OPEN_DIALOG({ id: PROJECT_COPY_DIALOG, data: {} }));
-    };
+import * as React from "react";
+import { InjectedFormProps, Field, WrappedFieldProps } from 'redux-form';
+import { WithDialogProps } from '~/store/dialog/with-dialog';
+import { FormDialog } from '~/components/form-dialog/form-dialog';
+import { ProjectTreePicker } from '~/views-components/project-tree-picker/project-tree-picker';
+import { Typography } from "@material-ui/core";
+import { COPY_NAME_VALIDATION, MAKE_A_COPY_VALIDATION } from '~/validators/validators';
+import { TextField } from "~/components/text-field/text-field";
+import { ProjectCopyFormDialogData } from "~/store/project-copy-dialog/project-copy-dialog";
 
-export const ProjectCopyDialog = compose(
-    withDialog(PROJECT_COPY_DIALOG),
-    reduxForm({
-        form: PROJECT_COPY_DIALOG,
-        onSubmit: (data, dispatch) => {
-            dispatch(startSubmit(PROJECT_COPY_DIALOG));
-            setTimeout(() => dispatch(stopSubmit(PROJECT_COPY_DIALOG, { name: 'Invalid path' })), 2000);
-        }
-    })
-)(ProjectCopy);
\ No newline at end of file
+export const ProjectCopyFormDialog = (props: WithDialogProps<string> & InjectedFormProps<ProjectCopyFormDialogData>) =>
+    <FormDialog
+        dialogTitle='Make a copy'
+        formFields={ProjectCopyFields}
+        submitLabel='Copy'
+        {...props}
+    />;
+
+const ProjectCopyFields = () => <div>
+    <ProjectCopyNameField />
+    <ProjectCopyDialogFields />
+</div>;
+
+const ProjectCopyNameField = () =>
+    <Field
+        name='name'
+        component={TextField}
+        validate={COPY_NAME_VALIDATION}
+        label="Enter a new name for the copy" />;
+
+const ProjectCopyDialogFields = () =>
+    <Field
+        name="projectUuid"
+        component={ProjectPicker}
+        validate={MAKE_A_COPY_VALIDATION} />;
+
+const ProjectPicker = (props: WrappedFieldProps) =>
```

Comment:
Could you extract ProjectPicker component as a part of this PR? Then it could be also reused in move dialogs.